tests/gflownet/envs/test_crystals.py

```diff
@@ -251,11 +251,13 @@ def test__step__does_not_change_state_if_element_already_set(
     assert valid
     assert initial_state != state_after_valid
 
-    state_after_invalid, action, valid = env.step(invalid_action)
 
-    assert action == invalid_action
-    assert not valid
-    assert state_after_valid == state_after_invalid
+# TODO: uncomment when step can handle invalid actions
+#     state_after_invalid, action, valid = env.step(invalid_action)
+
+#     assert action == invalid_action
+#     assert not valid
+#     assert state_after_valid == state_after_invalid
```

Comment:
A bit of a general question on this PR... how damageable (or not) is it that the step() function doesn't ensure that the incoming action is valid anymore?
Was the CategoricalSampling the only possible source of invalid actions?
If all actions are sampled using the mask then I think it should be fine but I don't know the code base well enough to garantee that it is the case.
For instance, if we can have actions entered by a user or actions specified by a replay buffer stored on disk (and which could, therefore, contain actions that were valid under a previous version of the env but not the current one), then there could be an issue.

Reply:
As far as I can recall, the current typical pipeline to train GFlowNet should never sample and execute actions without a mask. However, the code allows using parts of it outside this pipeline and future code may call the step function without having computed a mask. This is why I am not fully satisfied with the removal of the check.

> Was the CategoricalSampling the only possible source of invalid actions?
Hard to tell.

Reply:
What do you think about the tasks added in the PR to sort this out? Namely:
- Implement "safe mode", instead of completely removing the mask check in step()
- Add info about "safe mode" in the README

Reply:
I agree, these two things would do a lot to mitigate the risk of removing this layer of security. I think it's a good idea.

Reply:
Done now